clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -280,7 +295,27 @@ void Sema::checkSYCLDeviceVarDecl(VarDecl *Var) {
   SourceRange Loc = Var->getLocation();
   llvm::DenseSet<QualType> Visited;
 
-  checkSYCLVarType(*this, Ty, Loc, Visited);
+  checkSYCLVarType(*this, Ty, Loc, Visited, false);
+}
+
+static void checkAccessorVarType(Sema &S, QualType Ty, SourceRange Loc) {
+  assert(Util::isSyclAccessorType(Ty) &&
+         "Should only be called on SYCL accessor types.");
+
+  llvm::DenseSet<QualType> Visited;
+  const RecordDecl *RecTy = Ty->getAsRecordDecl();
+  for (const auto &Field : RecTy->fields()) {
+    if (Field->getName().compare("MData") == 0) {
+      QualType FTy = Field->getType();
+      // Sub-reference array or pointer, then proceed with that type.
```

Comment:
BTW, do we really have to prohibit pointers to unsupported types?

Reply:
It's just how the accessor is modelled.  The MData field of the accessor is a pointer to the data. We want to know the type of that data (pointee) so we can check it. 
Did I answer your question? I may be misunderstanding what you are asking.

Reply:
Yeah, I know that MData field is a pointer. But AFAIK it is not a pointer to pointer and etc. Here and in your previous patches you are taking `getPointeeOrArrayElementType` from the pointer type pointer unless it is not a pointer, so if there was appearance of pointer to unsuppoted type in device code, this is diagnosed as well.
Do pointers to unsupported types break SPIR-V translation as well? i.e.
```
accessor<__float128 *, ...> acc;

/// DEVICE CODE

// does this
acc.use();
// or this breaks SPIR-V translation?
__float128 *A;

Reply:
ah, I understand. Yes, this breaks the SPIR-V translation. Just double checked it. Will add a test for this.

Reply:
test added